src/Workspaces/Core/Portable/CodeGeneration/ICodeGenerationService.cs

```diff
@@ -175,5 +175,13 @@ internal interface ICodeGenerationService : ILanguageService
         /// <c>true</c> if destination is a location where other symbols can be added to.
         /// </summary>
         bool CanAddTo(SyntaxNode destination, Solution solution, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Return the most relevant declaration to namespaceOrType,
+        /// it will first search the context node contained within,
+        /// then the declaration in the same file, then non auto-generated file,
+        /// then all the potential location. Return null if no declaration.
+        /// </summary>
+        Task<SyntaxNode> FindMostRelevantNameSpaceOrTypeDeclarationAsync(Solution solution, INamespaceOrTypeSymbol namespaceOrType, CodeGenerationOptions options = null, CancellationToken cancellationToken = default);
```

Comment:
Don't take optional arguments, you'll forget to pass them along (which is what happened in this PR).